woocommerce/woocommerce-gutenberg-products-block
# Remove CustomerDataContext

Due to recent changes (data [store refactor](https://github.com/woocommerce/woocommerce-blocks/pull/6612) and #5817), the customer data context has become redundant. What was previously stored here now lives in the cart store and can be accessed directly. This PR removes the CustomerDataContext and associated hook - `useCustomerData()`.




Fixes #6108 

#### User Facing Testing



The following functionality requires smoke testing:

1. Checking out - edit shipping and billing sections, confirm things persist, confirm orders place with correct data
2. Paying via express payment methods. Confirm they can still set and update addresses.
3. Confirm updates to the server are debounced (500ms intervals) when editing address data

User Testing instructions

1. Add something to cart and go to checkout, open the network tab of your dev tools.
2. Enter a shipping address - watch for XHR requests to the batch endpoint. You should only get one request one second after you last made changes. If you make multiple changes all within one second of each other, there should still only be one request. Ensure multiple requests for the same change don't appear.
3. Uncheck "use shipping as billing"
4. Enter a different billing address
5. Pay with credit card via stripe
6. Check the shipping and billing addresses on the order are correct.
7. Open the Checkout Block in the editor. Check all shipping and billing address fields have an empty address.
8. Click the `Use same address for billing` checkbox. Ensure the billing address shows and hides when you tick this box


### WooCommerce Visibility
